### PR TITLE
pre-opp

### PR DESCRIPTION
**Pull request by AnnaStoyano**
### Pre-oop

[Demo](https://annastoyano.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/AnnaStoyano/a-tiny-JS-world)

Please, review.

### CODE REVIEW

File: submissions/AnnaStoyano/pre-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+};
+
+
+const inhabitants = [man,woman,cat,dog,catWoman];
+let output = '';
+inhabitants.forEach(item=>{
+   for(let key in item){
+      output+=`<b>${key}</b>: ${item[key]}\t`
+   }
+   output+='\n'
```
Comment (OleksiyRudenko) [inline submissions/AnnaStoyano/pre-oop/index.js:67]: This fragment relies on:
1. order of object properties (which is not guaranteed)
2. there are no extra properties we do not want in output

Both issues can be resolved by specifying the list of keys and mapping it into `<b>${key}</b>: ${item[key]}\t`.

Besides, after having keys mapped into strings they can be joined with `\t`, and strings per object then subsequently can be joined with `\n`.

In other words you might use `.map`, `.join` also nested (map+join per each object, map+join per key list for each object).

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@AnnaStoyano Well done!
Please let me know if you want to take a challenge as per comment below.
The purpose is to improve code scalability and maintainability by using `Array` methods (in this particular case).

**Comment by AnnaStoyano** [comment]
Thank you for your advice)

File: submissions/AnnaStoyano/pre-oop/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+const inhabitants = [man,woman,cat,dog,catWoman];
+
+const properties = ['name','species','gender','legs','hands','saying','friends'];
+let output = inhabitants.map(inhabitant=>{
+   let match = properties.map(property=>{
+      return `<b>${property}:</b> ${inhabitant[property]}`
+   });
+   return match.join('\t')
+})
```
Comment (OleksiyRudenko) [inline submissions/AnnaStoyano/pre-oop/index.js:69]: Chaining:
```
let a = arr.map(...);
return a.join(...);

// versus
return arr.map(...).join(...);
```

Arrow functions:
```
arr.map(e => {
  let a = e*2;
  return a;
})

// versus
arr.map(e => e*2);
```
Please let me know if the above gives you hints on how to make the code even more concise.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
Good job!
One more iteration to make it great

**Review by OleksiyRudenko — APPROVED** [review]
@AnnaStoyano Great job!

**Pull request merged**